ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java

```diff
@@ -137,9 +159,19 @@ private void addTruncateTableOutputs(ASTNode root, Table table, Map<String, Stri
       }
     } else {
       if (AlterTableUtils.isFullPartitionSpec(table, partitionSpec)) {
-        validatePartSpec(table, partitionSpec, (ASTNode) root.getChild(1), conf, true);
-        Partition partition = PartitionUtils.getPartition(db, table, partitionSpec, true);
-        outputs.add(new WriteEntity(partition, writeType));
+        if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
```

Comment:
em, i don't think iceberg would be `alwaysUnpartitioned`

Reply:
IcebergHandler is always unpartitioned, also after satisfying these clauses, we call `getPartitionsByFilter`.

Reply:
it unpartitioned because we don't have support for that at this moment, it won't be always unpartitioned

Reply:
I agree, but if you dont have alwaysUnpartitioned here, are you suggesting making a `storageHandler.getPartitionsByFilter` call from `getPartitions` API?

Reply:
yea, i think that would be more cleaner approach

Reply:
> I agree, but if you dont have alwaysUnpartitioned here, are you suggesting making a storageHandler.getPartitionsByFilter call from getPartitions API?

Implemented this. Done.